homeassistant/components/mqtt/mixins.py

```diff
@@ -274,6 +273,22 @@ def validator(config: ConfigType) -> ConfigType:
     return validator
 
 
+@dataclass
+class MqttData:
+    """Keep the MQTT entry data."""
+
+    client: MQTT | None = None
+    config: ConfigType | None = None
+    device_triggers: dict[str, Trigger] = field(default_factory=dict)
+    discovery_registry_hooks: dict[tuple, CALLBACK_TYPE] = field(default_factory=dict)
```

Comment:
```suggestion
    discovery_registry_hooks: dict[tuple[str, str], CALLBACK_TYPE] = field(default_factory=dict)
```

You will also need to change this then.
https://github.com/home-assistant/core/blob/7eefaa308f2ba083f37e1841889b05f5aedbf371/homeassistant/components/mqtt/mixins.py#L900
```diff
-discovery_hash: tuple
+discovery_hash: tuple[str, str]
```